src/cpp/include/openvino/genai/llm_pipeline.hpp

```diff
@@ -207,7 +211,7 @@ class OPENVINO_GENAI_EXPORTS LLMPipeline {
 
     void start_chat();
     void finish_chat();
-    std::string apply_chat_template(std::string prompt, std::string role = "user") const;
+    std::string apply_chat_template(const ChatHistory& history, bool add_generation_prompt=true) const;
```

Comment:
how users can use this method with chat mode?

Reply:
To make similar with HF and moved `apply_chat_template` to tokenizer, since in concerns only string manipulation.

About feeding a concatenated input, it turned out that this not so trivial in chat scenario (after `start_chat`), because it that case user would have to apply template and only after that call generate but the very reason we have `start_chat` is that since history is kept in KV cache we can get from user only the last raw prompt and do everything else for ourselves.

So answer to your question is, user don't have to call it manually when he did `start_chat`.

In case when history is kept in strings, (no `start_chat` is called), then he will have to accumulate history in ChatHistory structure List[{"role": role, "content": "whatever prompt"}], and the lates one is almost to HF and is presented in out tests. Examples below

**With `start_chat`:**
```
chat_history_ov = []
pipe.start_chat()
tok = pipe.get_tokenizer()
for prompt in quenstions:
	chat_history_ov.append({'role': 'user', 'content': prompt})
	
	answer_ov = pipe.generate(prompt, **config)
	chat_history_ov.append({'role': 'assistant', 'content': answer_ov}, add_generation_prompt=False)
pipe.finish_chat()
```

**Without `start_chat`**, difference it that we have to call `apply_chat_template` on the whole history and feed `chat_prompt_full_hist`:
```
chat_history_ov = []
tok = pipe.get_tokenizer()
for prompt in quenstions:
	chat_history_ov.append({'role': 'user', 'content': prompt})
	
	chat_prompt_full_hist = tok.apply_chat_template(chat_history_ov, add_generation_prompt=True)
	tokenized = tok(chat_prompt)
	
	answer_ov = pipe.generate(chat_prompt_full_hist, **config)
	chat_history_ov.append({'role': 'assistant', 'content': answer_ov}, add_generation_prompt=False)
```

**HF:**
```
chat_history_hf = []
for prompt in quenstions:
	chat_history_hf.append({'role': 'user', 'content': prompt})
	
	chat_prompt = tokenizer.apply_chat_template(chat_history_hf, tokenize=False, add_generation_prompt=True)
	tokenized = tokenizer(chat_prompt, return_tensors='pt', add_special_tokens=False)
	
	answer = model_opt.generate(**tokenized, **config_hf)
	answer_str = tokenizer.decode(answer[0, tokenized['input_ids'].numel():], skip_special_tokens=True)
	chat_history_hf.append({'role': 'assistant', 'content': answer_str})
```